src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -243,6 +243,13 @@ private static void ThrowServerException(HttpResponse response, RequestContext r
 
             if (shouldLogAsError)
             {
+                requestContext.Logger.ErrorPii(
+                    string.Format(MsalErrorMessage.RequestFailureErrorMessagePii,
+                        requestContext.ApiEvent?.ApiIdString,
+                        requestContext.ServiceBundle.Config.Authority.AuthorityInfo.CanonicalAuthority,
+                        requestContext.ServiceBundle.Config.ClientId), 
+                    string.Format(MsalErrorMessage.RequestFailureErrorMessage,
```

Comment:
Maybe also add `requestContext.ServiceBundle.Config.Authority.AuthorityInfo.Host` to non-PII message.